File: main/src/com/seovic/core/extractor/ScriptExtractor.java
/*
 * Copyright 2009 Aleksandar Seovic
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.seovic.core.extractor;


import com.seovic.core.Extractor;
import com.seovic.core.expression.ScriptExpression;


/**
 * An imlementation of {@link Extractor} that extracts value from a target
 * object using {@link ScriptExpression}.
 *
 * @author Aleksandar Seovic  2009.11.07
 */
public class ScriptExtractor
        extends ExpressionExtractor
    {
    // ---- constructors ----------------------------------------------------

    /**
     * Deserialization constructor (for internal use only).
     */
    public ScriptExtractor()
        {
        }

    /**
     * Construct an <tt>ScriptExtractor</tt> instance.
     *
     * @param expression  the expression to use
     */
    public ScriptExtractor(String expression)
        {
        super(new ScriptExpression(expression));
        }

    /**
     * Construct an <tt>ScriptExtractor</tt> instance.
     *
     * @param expression  the expression to evaluate
     * @param language    scripting language to use
     */
    public ScriptExtractor(String expression, String language)
        {
        super(new ScriptExpression(expression, language));
        }
    }

File: trunk/main.net/src/Coherence.Tools/Core/Extractor/ScriptExtractor.cs
using System.IO;
using Seovic.Core.Expression;

namespace Seovic.Core.Extractor
{
    /// <summary>
    /// An imlementation of <see cref="IExtractor"/> that extracts value 
    /// from a target object using <see cref="ScriptExpression"/>.
    /// </summary>
    public class ScriptExtractor : ExpressionExtractor
    {
        /// <summary>
        /// Deserialization constructor (for internal use only).
        /// </summary>
        public ScriptExtractor()
        {
        }

        /// <summary>
        /// Construct an <c>ScriptExtractor</c> instance.
        /// </summary>
        /// <param name="expression">The expression to evaluate.</param>
        public ScriptExtractor(string expression)
            : base(new ScriptExpression(expression))
        {
        }

        /// <summary>
        /// Construct an <c>ScriptExtractor</c> instance.
        /// </summary>
        /// <param name="script">The script to evaluate.</param>
        public ScriptExtractor(TextReader script)
            : base(new ScriptExpression(script))
        {
        }

        /// <summary>
        /// Construct an <c>ScriptExtractor</c> instance.
        /// </summary>
        /// <param name="expression">The expression to evaluate.</param>
        /// <param name="language">Scripting language to use.</param>
        public ScriptExtractor(string expression, string language)
            : base(new ScriptExpression(expression, language))
        {
        }

        /// <summary>
        /// Construct an <c>ScriptExtractor</c> instance.
        /// </summary>
        /// <param name="script">The script to evaluate.</param>
        /// <param name="language">Scripting language to use.</param>
        public ScriptExtractor(TextReader script, string language)
            : base(new ScriptExpression(script, language))
        {
        }
    }
}